File: android/app/src/main/java/com/aitutor/app/domain/repository/AuthRepository.kt
package com.aitutor.app.domain.repository

import com.aitutor.app.domain.model.AuthState
import kotlinx.coroutines.flow.Flow

interface AuthRepository {
    val authState: Flow<AuthState>

    suspend fun signInWithEmail(email: String, password: String): Result<Unit>
    suspend fun signUpWithEmail(email: String, password: String, displayName: String?): Result<Unit>
    suspend fun signInWithGoogle(): Result<Unit>
    suspend fun signOut(): Result<Unit>
    suspend fun sendPasswordReset(email: String): Result<Unit>
}

File: ios/AITutor/Domain/Repository/AuthRepository.swift
import Foundation

protocol AuthRepository: AnyObject {
    /// Async stream of session changes — emits .unknown while loading, then
    /// .signedIn / .signedOut as auth state mutates.
    var authState: AsyncStream<AuthState> { get }

    func signInWithEmail(email: String, password: String) async throws
    func signUpWithEmail(email: String, password: String, displayName: String?) async throws
    func signInWithGoogle() async throws
    func signOut() async throws
    func sendPasswordReset(email: String) async throws
}
